modules/nf-core/seqtk/sample/main.nf

```diff
@@ -0,0 +1,58 @@
+process SEQTK_SAMPLE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqtk:1.4--he4a0461_1' :
+        'biocontainers/seqtk:1.4--he4a0461_1' }"
+
+    input:
+    tuple val(meta), path(reads), val(sample_size)
+
+    output:
+    tuple val(meta), path("*.fastq.gz"), emit: reads
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if (!(args ==~ /.*-s[0-9]+.*/)) {
+        args += " -s100"
+    }
+    if ( !sample_size ) {
```

Comment:
If you think that adding support for relative sample sizes is desirable, you could use the [countFastq()](https://www.nextflow.io/docs/latest/reference/operator.html#countfastq) operator to determine the number of records quickly. Once you established the number of reads in your input, you could then use a relative parameter to calculate the final size of the subset.

But that would be a stretch goal, I think.

Reply:
I have open this issue to add it later: https://github.com/nf-core/seqinspector/issues/60